### DISCUSSION TITLE
Uploading a large library to immich without crashing

### DISCUSSION THREAD

**Discussion by alexlyee** [immich-app/immich#2773] [General]
Hey, I'm trying to upload a very large collection of my photos to immich from my NAS but it's crashing the browser while doing so. I'm wondering if an alternative could be or is to just drop the photos into the uploads folder in some way? It would take me many painstaking hours to get these photos/videos uploaded in small sections at a time.

**Comment by bo0tzz** [comment]
What you want is the CLI: https://immich.app/docs/features/bulk-upload

**Comment by alextran1502** [comment]
You can use the CLI to handle this situation
